server/src/main/java/org/opensearch/index/translog/Translog.java

```diff
@@ -1745,40 +1746,42 @@ public void trimUnreferencedReaders() throws IOException {
             }
         }
 
-        // move most of the data to disk to reduce the time the write lock is held
-        sync();
-        try (ReleasableLock ignored = writeLock.acquire()) {
-            if (closed.get()) {
-                // we're shutdown potentially on some tragic event, don't delete anything
-                return;
-            }
-            final long minReferencedGen = getMinReferencedGen();
-            for (Iterator<TranslogReader> iterator = readers.iterator(); iterator.hasNext();) {
-                TranslogReader reader = iterator.next();
-                if (reader.getGeneration() >= minReferencedGen) {
-                    break;
+        synchronized (uploadMutex) {
```

Comment:
trimming is already under `writeLock` . Why do we need a mutex here ?

Reply:
Removed this.